pkg/sensors/program/loader_linux.go

```diff
@@ -692,13 +536,59 @@ func LoadUprobeProgram(bpfDir string, load *Program, maps []*Map, verbose int) e
 func LoadMultiKprobeProgram(bpfDir string, load *Program, maps []*Map, verbose int) error {
 	opts := &LoadOpts{
 		Attach: MultiKprobeAttach(load, bpfDir),
-		Open:   KprobeOpen(load),
-		Maps:   maps,
+		Open: func(coll *ebpf.CollectionSpec) error {
+			return nil
+		},
+		Maps: maps,
 	}
 	return loadProgram(bpfDir, load, opts, verbose)
 }
 
-func LoadFmodRetProgram(bpfDir string, load *Program, maps []*Map, progName string, verbose int) error {
+func LoadKProbeOverrideProgram(bpfDir string, load *Program, maps []*Map, progName string, verbose int, unloadFunc func(bool) error) error {
```

Comment:
hm why do we need special override function for kprobe and not for fmodret? can't we just use LoadKprobeProgram?

Reply:
Thanks for the feedback!  In the current code base, in addition to override programs, `LoadKprobeProgram()` loads kprobe programs used in kprobe sensors and `LoadFmodRetProgram()` loads the override programs for enforcer, so it was an attempt to ensure that the `unloaderOverride()` is only specified and called in the override program scenario.

Combining all the comments together, now I think it makes more sense to move the `genericoverride.go` to under `github.com/cilium/tetragon/pkg/sensors/program`.  This way I can assign unexported `unloaderOverride` there, and then I don't need many changes in this version.

Thanks for your feedback again!